rules/no-hex-escape.js

```diff
@@ -1,25 +1,21 @@
 'use strict';
 
-function checkEscape(context, node, value) {
-	const fixedValue = typeof value === 'string' ? value.replace(/\\x/g, '\\u00') : value;
+const create = context => {
+	const checkEscape = (node, value) => {
+		const fixedValue = typeof value === 'string' ? value.replace(/\\x/g, '\\u00') : value;
 
-	if (value !== fixedValue) {
-		context.report({
-			node,
-			message: 'Use unicode escapes instead of hexadecimal escapes.',
-			fix: fixer => fixer.replaceTextRange([node.start, node.end], fixedValue)
-		});
-	}
-}
+		if (value !== fixedValue) {
+			context.report({
+				node,
+				message: 'Use unicode escapes instead of hexadecimal escapes.',
+				fix: fixer => fixer.replaceTextRange([node.start, node.end], fixedValue)
+			});
+		}
+	};
 
-const create = context => {
 	return {
-		Literal: node => {
-			checkEscape(context, node, node.raw);
-		},
-		TemplateElement: node => {
-			checkEscape(context, node, node.value.raw);
-		}
+		Literal: node => checkEscape(node, node.raw),
+		TemplateElement: node => checkEscape(node, node.value.raw)
```

Comment:
You could enforce that with a type checker like Flow/TS, though a bit troublesome.

I'm starting to think about having a plugin to write rules, not sure what I'd put in there yet. Having Flow types for the whole AST would be so much more useful though, so thinking about it (but not doing much ATM :sweat_smile:).